.github/workflows/ci.yml

```diff
@@ -3,27 +3,29 @@ name: CI
 on: [push]
 
 jobs:
-
   test:
-    runs-on: ubuntu-latest
+    name: "Python ${{ matrix.python-version }}"
+    runs-on: "ubuntu-latest"
+    env:
+      USING_COVERAGE: '3.8'
 
-    steps:
-      - uses: actions/checkout@v1
-        with:
-          fetch-depth: 1
+    strategy:
+      matrix:
+        python-version: ["3.6", "3.7", "3.8"]
 
-      - name: Set up Python 3.7
-        uses: actions/setup-python@v1
+    steps:
+      - uses: "actions/checkout@v2"
+      - uses: "actions/setup-python@v1"
         with:
-          python-version: 3.7
-
-      - name: Setup Dependencies
+          python-version: "${{ matrix.python-version }}"
+      - name: "Install dependencies"
         run: |
-          python -m pip install --upgrade pip
-          pip install -r requirements_dev.txt
-
-      - name: Code Quality
-        run: black --check --diff .
+          set -xe
+          python -VV
+          python -m site
+          python -m pip install --upgrade pip setuptools wheel
+          python -m pip install --upgrade coverage[toml] virtualenv tox tox-gh-actions
+          python -m pip install -r requirements_dev.txt --upgrade 
```

Comment:
I'm confused...you're installing tox but not running it.

Reply:
Yeah I didn't cut and paste enough and hadn't gotten around to looking closer. It looks like everything works now.

I'm a little bit fuzzy on how Coveralls is picking which build is "canonical" for the commit (first reported? last? all? any?) but it probably doesn't matter in practice.